framework/src/predictors/AdamsPredictor.C

```diff
@@ -73,6 +73,10 @@ AdamsPredictor::apply(NumericVector<Number> & sln)
   // that gets called on time step begin.
   // That means that history control must go here.
   historyControl();
+
+  if (!shouldApply())
```

Comment:
That looks really easy to do. We could add a ```timestepSetup()``` method to the predictor that gets called from ```NonlinearSystem::onTimestepBegin()```, and move everything in ```AdamsPredictor::historyControl()``` into that method.  Once we did that, we could have the framework call ```shouldApply()``` and only call ```apply()``` if that's true.  I'll go ahead and do that unless I hear otherwise.